### PR TITLE
Bratseth/pack bits

### PR DESCRIPTION
**Pull request by bratseth**
@geirst please review. 

Sorry for the many commits, this was a bit exploratory.

Now that we have contextual type resolving, I want to replace the earlier half-baked attempts at type resolving by it, but this diff is already large enough.

### CODE REVIEW

**Comment by bratseth** [comment]
No rush, but have you seen this @geirst ?

**Comment by bratseth** [comment]
Take your time @geirst ... :-)

File: indexinglanguage/src/main/java/com/yahoo/vespa/indexinglanguage/expressions/AnyDataType.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package com.yahoo.vespa.indexinglanguage.expressions;
```
Comment (geirst) [inline indexinglanguage/src/main/java/com/yahoo/vespa/indexinglanguage/expressions/AnyDataType.java:1]: Please add copyright.

**Review by geirst — DISMISSED** [review]
👍 I am not sure if the new input / output type resolving in StatementExpression.doVerify() is tested by existing unit tests or if new tests should be added (especially error situations).

**Comment by bratseth** [comment]
Thanks! Yes, more work is needed on this type resolving for sure. This is just the bare minimum to support both embed and pack_bits in the same statement.

**Pull request merged**